src/main/java/io/github/josiasbarreto_dev/desafio_naruto/service/CharacterService.java

```diff
@@ -0,0 +1,154 @@
+package io.github.josiasbarreto_dev.desafio_naruto.service;
+
+import io.github.josiasbarreto_dev.desafio_naruto.dto.ChakraRequestDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.dto.CharacterRequestDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.dto.CharacterResponseDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.dto.JutsuRequestDTO;
+import io.github.josiasbarreto_dev.desafio_naruto.exception.ResourceNotFoundException;
+import io.github.josiasbarreto_dev.desafio_naruto.mapper.CharacterMapper;
+import io.github.josiasbarreto_dev.desafio_naruto.model.*;
+import io.github.josiasbarreto_dev.desafio_naruto.model.Character;
+import io.github.josiasbarreto_dev.desafio_naruto.repository.GenjutsuRepository;
+import io.github.josiasbarreto_dev.desafio_naruto.repository.NinjutsuRepository;
+import io.github.josiasbarreto_dev.desafio_naruto.repository.TaijutsuRepository;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CharacterService {
+    private final GenjutsuRepository genjutsuRepository;
+    private final NinjutsuRepository ninjutsuRepository;
+    private final TaijutsuRepository taijutsuRepository;
+
+    private final CharacterMapper mapper;
+
+    public CharacterService(GenjutsuRepository genjutsuRepository, NinjutsuRepository ninjutsuRepository, TaijutsuRepository taijutsuRepository, CharacterMapper mapper) {
+        this.genjutsuRepository = genjutsuRepository;
+        this.ninjutsuRepository = ninjutsuRepository;
+        this.taijutsuRepository = taijutsuRepository;
+        this.mapper = mapper;
+    }
+
+    public CharacterResponseDTO createCharacter(CharacterRequestDTO requestDTO){
+        switch (requestDTO.ninjaType()) {
+            case GENJUTSU:
+                var genjutsuToSave = mapper.toEntityGenjutsu(requestDTO);
+                GenjutsuNinja savedCharacterGenjutsu = genjutsuRepository.save(genjutsuToSave);
+                return mapper.toDTO(savedCharacterGenjutsu);
+
+            case NINJUTSU:
+                var ninjutsuToSave = mapper.toEntityNinjutsu(requestDTO);
+                NinjutsuNinja savedCharacterNinjutsu = ninjutsuRepository.save(ninjutsuToSave);
+                return mapper.toDTO(savedCharacterNinjutsu);
+
+            case TAIJUTSU:
+                var taijutsuToSave = mapper.toEntityTaijutsu(requestDTO);
+                TaijutsuNinja savedCharacterTaijutsu = taijutsuRepository.save(taijutsuToSave);
+                return mapper.toDTO(savedCharacterTaijutsu);
+
+            default:
+                return null;
```

Comment:
Caso caia nesta condição pode causar um nullPointerException, o recomendado é que seja lançado uma exceção para quando o tipo de ninja seja inválido, segue uma sugestão:
```suggestion
                return throw new IllegalArgumentException("tipo de ninja é inválido");
```